### ISSUE TITLE
Linux Mint 22/Ubuntu 24.04 python library error

### ISSUE THREAD

**Issue by kd8bxp** [Andy4495/TI_Platform_Cores_For_Arduino#6]
I'm not sure you can help with this issue, I'm not sure it's an issue of the core or not. 
but I get this error when trying to upload on Linux Mint 22
/home/lfmiller/.arduino15/packages/energia/tools/dslite/9.3.0.1863/DebugServer/bin/DSLite: error while loading shared libraries: libpython2.7.so.1.0: cannot open shared object file: No such file or directory

I'm pretty sure this is because Python 2.7.x has been deprecated, and dslite 9.3.0.1863 requires it.
I haven't been able to find any kind of work around for the issue. 

Hoping maybe you know of something that might work?

Thanks in advance.

**Comment by Andy4495** [comment]
I can take a look, but I just copied the board packages as-is, and didn't create the dislite tool, so I can't guarantee anything.

Have you tried installing python2 manually: 

```
apt install python2
```

Let me know if that works, and I'll post an update here if I can come up with any alternate solutions.

**Comment by kd8bxp** [comment]
Unfortunately that was one of the first things I did try. 

And of course that gives the error
`E: Package 'python2' has no installation candidate`

When you say copied the board packages as-is? Do you mean putting manually installing it IE: using the hardware directory? and not using the board manager?

**Comment by Andy4495** [comment]
You can try to use TI's Uniflash tool: https://www.ti.com/tool/UNIFLASH

You can manually run it to upload the binary created from Arduino.

There is a way to have Uniflash to export everything into a package which includes dslite and all the config files you need for specific boards.

The last time I used Uniflash was several years back when TI released the MSP430-G2ET boards (see https://github.com/Andy4495/MSP-EXP430G2ET-Energia) -- the G2ET boards didn't work with Energia at the time, so I used it until and updated Energia was released. 

I'll try and see if I can get Uniflash to create an updated dslite that doesn't depend on Python2 tool and make the appropriate updates to the board package files to get this to work, but it will take a few days. 

[My reference to using the packages "as-is", I meant that I copied the board packages to my GitHub repo as-is and didn't modify the internals of the board packages and tools except to clean up the board package JSON files with up-to-date URLs. In the Arduino IDE, I install them using the board manager.]

**Comment by kd8bxp** [comment]
Thank you!!!
This worked, sort of.
The Uniflash tool did install (it gave an error when it first started, but processed to work and install).

What I did thou:
First I compiled my sketch, and exported the bin file.
Next I started the UniFlash tool, selected my board.
Clicked on "start"
That gave me this screen
![Screenshot at 2024-08-15 19-55-03](https://github.com/user-attachments/assets/35c75206-3662-4519-b213-44127960b667)

Where I was able to select the .bin file I just created.  Plugged the board in, and clicked on load image.

That worked (it updated the firmware of the board first) - But it seems like that process worked with out any issues.
I was able to create a dslite.sh script for standalone, I haven't tried it yet.

I'm pretty happy this worked even thou it does take a couple of extra steps, it's a fair work around to the issue.

Thanks again!
-LeRoy

**Comment by kd8bxp** [comment]
Just to update, I couldn't get the stand-alone programmer to work - It's me, I'm sure I'm not understanding what it wants as input. 
BUT I want to thank you @Andy4495  for your effort to preserve this.  I've had one of the RSLK Robot kits sitting brand new in a box for years - decided to pull it out and play around with it. Only to find out that most of the information about is gone.

Then I found your repository, then found I could compile, but not upload the code.  The Uniflash did the trick, and will keep this alive a while longer!

Your efforts have helped me in so many ways. 
And have saved me from setting up a machine just to play with the robot.

I think there is enough example code now to at least piece together how this robot works and maybe make something fun and (not so) useful. 
Thank you! Keep up the good work! 

-LeRoy

**Comment by Andy4495** [comment]
LeRoy - 

I finally got around to creating a new dslite tool package and updated the board packages to use the newer dslite. I have tested this on Ubuntu 24, 22, Windows, and Mac, so I think this should also work with Linux Mint 22.

If you use this URL in the Board Manager: 

```text
https://raw.githubusercontent.com/Andy4495/TI_Platform_Cores_For_Arduino/main/json/package_energia_optimized_index.json`
```

And update to the latest board package version (1.1.0 for MSP430 or Tiva, or 5.30.0 for MSP432), then Arduino should load the updated dslite tool and it should work directly instead of having to use the separate UniFlash tool. 

Let me know if you have a chance to test it on Linux Mint, and if you are still having trouble, open a new issue. 

Andy
